Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/WorkspaceService.java

```diff
@@ -174,4 +173,75 @@
      * @throws Exception
      */
     Folder moveFolder(Integer folderId, Integer newParentFolderId) throws Exception;
+
+    /**
+     * Get list of Group and AccessType pair of Log
+     * @param logId Log ID
+     * @return list of Group and AccessType pair
+     */
+    Map<Group, AccessType> getLogACL(Integer logId);
+
+    /**
+     * Get list of Group and AccessType pair of Log
+     * @param processId Process ID
+     * @return list of Group and AccessType pair
+     */
+    Map<Group, AccessType> getProcessACL(Integer processId);
+
+    /**
+     * Get list of Group and AccessType pair of Log
+     * @param folderId Folder ID
+     * @return list of Group and AccessType pair
+     */
+    Map<Group, AccessType> getFolderACL(Integer folderId);
+
+    /**
```

Comment:
These XXXPermissions and XXXACL should be put in a separate class which is responsible for security/access right management. The WorkspaceService is responsible for managing the workspace such as create/remove/move folder, get list of subfolders, breadcrumbs, etc. The general principle is "one class one responsiblity". Access rights/permissions belong to a different type of responsiblity. The reason we need to care about responsibility in class design is because the class would be modified when its responsiblity changes, and thus, keep responsiblities in separate classes would help keep the code clean and minimal impact on the code when changes happen (always happen).